docs/sphinx/examples/python/tutorials/quantum_teleporation.ipynb

```diff
@@ -0,0 +1,402 @@
+    "m1 = kernel.mz(qubits[0])  \n",
+    "m2 = kernel.mz(qubits[1])  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#psi4 - apply conditioned pauli operators dependent on the measurement result of qubits 0 and 1 \n",
+    "\n",
+    "def x_condition(): \n",
+    "    kernel.x(qubits[2])\n",
+    "    \n",
+    "def z_condition(): \n",
+    "    kernel.z(qubits[2])\n",
+    "\n",
+    "kernel.c_if(m2, x_condition)\n",
+    "kernel.c_if(m1, z_condition)\n",
+    "\n",
+    "snapshot4 = cudaq.get_state(kernel) #store the state of the system for later use "
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The state of this 3 qubit system is described by a $2^3 = 8$ dimensional vector. Sometimes we would like to extract the state of a single qubit from a multipartite state. We can use the partial trace operation defined in the function below which allows us to translate our statevectors to the density matrix representation to trace out subsystems. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Function to be used below to calculate the partial trace yielding a density matrix \n",
+    "\n",
+    "def partial_trace(state_vector, trace_systems, dimensions):\n",
+    "    \"\"\"\n",
+    "    Partial trace of multi-particle quantum state. \n",
+    "\n",
+    "    Arguments:\n",
+    "        state_vector: complex vector  of size 2**n\n",
+    "        trace_systems (list(int)): a list of subsystems (starting from 0) to trace over. Note that we follow the MSB notation meaning the 0th qubit is on right. \n",
+    "        dimensions (list(int)): a list of the dimensions of the subsystems.\n",
+    "  \n",
+    "    Returns:\n",
+    "        ndarray: A density matrix with the appropriate subsystems traced over.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    rho = state_vector.reshape(dimensions)\n",
+    "    rho = np.tensordot(rho, rho.conj(), axes=(trace_systems, trace_systems))\n",
+    "    d = int(np.sqrt(np.prod(rho.shape)))\n",
+    "\n",
+    "    return rho.reshape(d, d)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "state_of_q0 = partial_trace(state_vector = np.array(snapshot0), trace_systems=[1,2], dimensions=[2,2,2]) #trace out qubits 1 and 2 leaving us with qubit 0\n",
+    "\n",
+    "state_of_q2 = partial_trace(state_vector = np.array(snapshot4), trace_systems=[0,1], dimensions=[2,2,2]) #trace out qubits 0 and 1 leaving us with qubit 2\n",
+    "\n",
+    "np.allclose(state_of_q0, state_of_q2)\n"
```

Comment:
nit: The output of these np.allclose's would look nicer, and self-document better, if they were wrapped in a print statement. Something like
```
print(f"|qubit_0> == |qubit_1>?  {np.allclose(state_of_q0, state_of_q1)}")
```